src/main/java/io/confluent/connect/hdfs/hive/HiveMetaStore.java

```diff
@@ -71,19 +71,10 @@ public HiveMetaStore(Configuration conf, HdfsSinkConnectorConfig connectorConfig
     R call() throws TException;
   }
 
-  private <R> R doWithRetry(ClientAction<R> action) throws TException {
-    try {
-      return action.call();
-    } catch (TException e) {
-      try {
-          client.reconnect();
-      } catch (MetaException swallowedException) {
-        // reconnect failed, throw the original exception
-        log.warn("Cannot connect to Hive meta store: {}", swallowedException.getMessage());
-        throw e;
-      }
-      return action.call();
-    }
+  private <R> R doAction(ClientAction<R> action) throws TException {
+    // No need to implement retries. We use RetryingHiveMetaStoreClient
+    // which already implements the retrying and reconnect logic
```

Comment:
Can you elaborate on what is the retry condition? It seems that two conditions can trigger retry: 1. The connection timeout is exceeded 2. We already retried and we should retry immediately. 

Also, do you know from which Hive version the `RetryingHiveMetaStoreClient` is included? Thanks!